### PR TITLE
tl detection without project_to_image_plane (zoom in) and a classifier

### PR DESCRIPTION
**Pull request by YCSU**
The __/traffic_waypoint__ topic now publishes the __index of the waypoint__ at which the car should stop. It publishes __-1__ when there is no traffic light in __red__ nearby, and publish the index for stopping when there is a traffic light in red near the car (now it is set to within 70 waypoints)

Since the classifier is not integrated yet, the state of the traffic light is obtained through __/vehicle/traffic_lights__.

### CODE REVIEW

**Comment by bmbigbang** [comment]
ok this might need some trial and error to get right. why have u removed variable assignments to the subscribers?
+        rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
+        rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)

I see a lot people trying to do things that are not really necessary. lets not do this... as the commits are quite bulky at the moment so hard to read  through them

**Comment by YCSU** [comment]
In the current situation, the variable assignments seem to be redundant since we use callback function to do what we need. With that said, I am not an expert in ROS. Is there any particular reason for the variable assignments?

BTW, I apologize for this bulky commit. The code was originally developed in my own repo hoping to minimize conflicts, and then copied it to the current repo.

**Comment by alex-n-braun** [comment]
Hi Yu-Cheng,

I think a file called traffic_light_config.py is missing. 

Cheers,
Alex

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -8,35 +8,40 @@
 from cv_bridge import CvBridge
 from light_classification.tl_classifier import TLClassifier
 import tf
+from tf.transformations import euler_from_quaternion
+import math
 import cv2
-import yaml
+import numpy as np
+from traffic_light_config import config
```
Comment (alex-n-braun) [inline ros/src/tl_detector/tl_detector.py]: this file seems to be missing.

**Review by alex-n-braun — CHANGES_REQUESTED** [review]
traffic_light_config.py is missing.

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -8,35 +8,40 @@
...
+        simulator. When testing on the vehicle, the color state will not be available. You'll need to
+        rely on the position of the light and the camera image to predict it.
         '''
-        sub3 = rospy.Subscriber('/vehicle/traffic_lights', TrafficLightArray, self.traffic_cb)
-        sub6 = rospy.Subscriber('/image_color', Image, self.image_cb)
 
-        config_string = rospy.get_param("/traffic_light_config")
-        self.config = yaml.load(config_string)
+        rospy.Subscriber('/vehicle/traffic_lights', TrafficLightArray, self.traffic_cb)
+        rospy.Subscriber('/camera/image_raw', Image, self.image_cb)
```
Comment (alex-n-braun) [inline ros/src/tl_detector/tl_detector.py]: why /camera/image_raw, and not /image_color?

**Comment by YCSU** [comment]
You are right! I checked the newest code from the Udacity repo. It looks like some part of my code comes from the older version. I will update it and upload it again tonight. Thanks!

**Pull request merged**